CONTRIBUTING.md

```diff
@@ -1,34 +1,34 @@
 # How to Contribute
 
-We'd love to accept your patches and contributions to this project. There are
-just a few small guidelines you need to follow.
-
-## Development Principles (for Googlers)
-
-There are a few principles for developing or refactoring the service
-implementations. Read the [Development Principles
-Guide](./docs/development-principles.md).
+Thank you so much for your interest in contributing to Online Boutique.
+Before contributing, you must:
+1. sign the [Contributor License Agreement (CLA)](#contributor-license-agreement).
+2. follow [Google's Open Source Community Guidelines](https://opensource.google.com/conduct/).
+3. follow the [Contribution Process](#contribution-process).
 
 ## Contributor License Agreement
 
-Contributions to this project must be accompanied by a Contributor License
-Agreement. You (or your employer) retain the copyright to your contribution;
-this simply gives us permission to use and redistribute your contributions as
+Contributions to Online Boutique must be accompanied by a Contributor License
+Agreement (CLA). You (or your employer) retain the copyright to your contribution.
+The CLA simply gives us permission to use and redistribute your contributions as
 part of the project. Head over to <https://cla.developers.google.com/> to see
 your current agreements on file or to sign a new one.
 
 You generally only need to submit a CLA once, so if you've already submitted one
 (even if it was for a different project), you probably don't need to do it
 again.
 
-## Code reviews
+## Contribution Process
 
-All submissions, including submissions by project members, require review. We
-use GitHub pull requests for this purpose. Consult
-[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
-information on using pull requests.
+Here's the process for making a change to this repository:
 
-## Community Guidelines
+1. If it's a **small change** (such as a bug fix or a spelling correction):
+    1. fork this repository and submit a [pull-request](https://help.github.com/articles/about-pull-requests/).
+    1. wait for a maintainer of this repository to review your change.
+1. If it's **not a small change**:
+    1. review Online Boutique's [purpose](/docs/purpose.md) and [product requirements](/docs/product-requirements.md).
+    1. if your change **does not align** with the purpose and product requirements, you should maintain your own fork of this repository.
+    1. if your change **does align**, create a [GitHub Issue](https://github.com/GoogleCloudPlatform/microservices-demo/issues/new/choose) describing the change **before** working on the implementation. It's important to create a GitHub Issue before working on the code to avoid wasting your development efforts.
+    1. wait for a maintainer of this repository to review your GitHub issue. If it's a large change, you might be asked to start a Google Doc to discuss design decisions.
```

Comment:
```suggestion
1. Review Online Boutique's [purpose](/docs/purpose.md) and [product requirements](/docs/product-requirements.md).
1. If your proposed changes **do not align** with the purpose and product requirements of Online Boutique, you may be asked to instead maintain your own fork of this repository.
1. For **small changes** (such as a bug fixes or spelling corrections):
    1. Fork this repository and submit a [pull request](https://help.github.com/articles/about-pull-requests/).
    1. Wait for a maintainer of this repository to review your change.
1. For **bigger changes**:
    1. Create a [GitHub issue](https://github.com/GoogleCloudPlatform/microservices-demo/issues/new/choose) describing the change **before** working on the implementation. This is important to avoid potentially having to discard your development efforts.
    1. Wait for a maintainer of this repository to review your GitHub issue. For significantly complex proposals, you may be asked to start a Google Doc to discuss design decisions.
```
Reordered / simplified (conceptually, even "small changes" could be against product requirements or product vision, so we can simply this section) + fixed a few grammatical / structural issues

Reply:
This is definitely cleaner.
And good point: even "small changes" could conflict with `product-requirements.md` and/or `purpose.md`.